src/error.rs

```diff
@@ -91,6 +102,17 @@ impl Error {
         }
     }
 
+    // Whether a write operation should be retried if this error occurs
+    pub(crate) fn is_write_retryable(&self) -> bool {
```

Comment:
Or--rather than making this private, tweak it to just check for the error label and introduce a new `should_add_retryable_write_label` or something that is private with this methods current body. The situation I'm trying to avoid with this is where we have different methods for determining if an error is retryable (e.g. in executor, we both check for the label and invoke this method in different places).

Reply:
Makes sense. I updated this to be `should_add_retryable_write_label` and added a different `is_write_retryable` to check for the label presence.